src/components/AbortionButton/AbortionButton.stories.js

```diff
@@ -0,0 +1,78 @@
+import {expect} from '@storybook/jest';
+import {waitFor, within} from '@storybook/testing-library';
+
+import {AnalyticsToolsDecorator} from 'story-utils/decorators';
+
+import AbortionButton from './AbortionButton';
+
+export default {
+  title: 'Private API / Abortion button',
+  component: AbortionButton,
+  decorators: [AnalyticsToolsDecorator],
+};
+
+export const Authenticated = {
+  args: {
+    isAuthenticated: true,
+  },
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    await waitFor(async () => {
+      const abortButton = await canvas.queryByRole('button', {name: 'Uitloggen'});
+      await expect(abortButton).toBeVisible();
+    });
```

Comment:
IMO the nicest is:

```js
const abortButton = await canvas.findByRole('button', {name: 'Uitloggen'});
await expect(abortButton).toBeVisible();
```
`findBy*` is really just a wrapper around `waitFor` + `getBy*`, but it makes the code flatter.

In general, you should use the async queries (`await findBy*`) when the component does stuff asynchronously, like `useEffect` or `useAsync` that may take some finite time to resolve. If the component is a "dumb" compnent and just instantly renders, typically the `getBy*` is preferred and can be done sync rather than async.